tests/installation/select_system_role.pm

```diff
@@ -0,0 +1,51 @@
+# Copyright (C) 2021 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+# Summary: Test module to handle System Role during installation:
+#          - Validate default system role selection (optional)
+#          - Select a different role if needed.
+#          - Navigate next screen
+
+# Maintainer: QE YaST <qa-sle-yast@suse.de>
+
+use strict;
+use warnings;
+use base 'y2_installbase';
+use scheduler 'get_test_suite_data';
+use testapi;
+use Test::Assert 'assert_equals';
+
+sub run {
+    my ($self) = @_;
+
+    my $test_data   = get_test_suite_data();
+    my $system_role = $testapi::distri->get_system_role_controller();
+
+    # check default role (in openSUSE could be optional due to none is preselected)
+    if (my $default = $test_data->{system_role}{default}) {
```

Comment:
I'm surprised that this syntax works, but I would put `->` as `$test_data->{system_role}` strictly speaking as an array reference and we do not do explicit conversion.

Reply:
I want to use in this way, in Perl we can omit multiple reference, it is just a matter of style, you can google it if you want, I noticed that in someone else's PR and I liked. It only has one con, if we would be mixing reference with no reference, but our test data have always references, so we can save that symbol when accessing multiple fields.